### PR TITLE
Sanitizer (dev).

### PR DESCRIPTION
**Pull request by fchirica**
- Make timelord able to generate compact proofs of time and n-wesolowski proofs of time (by using 'fast_algorithm' and 'sanitizer_mode' flags in config).
- Full node periodically broadcasts non compact proofs of time to timelords.
- Simulation also checks both nodes have at least one compact block at the end.
- Use 'create_periodic_introducer_poll_task' in tests (full_node_setup) to properly connect to the local introducer, if it exists.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
     "aiosqlite==0.13.0",  # asyncio wrapper for sqlite, to store blocks
     "aiohttp==3.6.2",  # HTTP server for full node rpc
     "colorlog==4.1.0",  # Adds color to logs
-    "chiavdf==0.12.16",  # timelord and vdf verification
+    "chiavdf==0.3b11",  # timelord and vdf verification
```
Comment (mariano54) [inline setup.py:13]: Why the strange version number?

File: src/full_node/full_node.py [hunk 2]
```diff
@@ -860,6 +866,81 @@ async def reject_compact_proof_of_time_request(
         for _ in []:
             yield _
 
+    async def broadcast_uncompact_blocks(
+        self, uncompact_interval, delivery: Delivery = Delivery.BROADCAST
+    ):
+        while self.sync_store.get_sync_mode():
```
Comment (mariano54) [inline src/full_node/full_node.py:872]: It it possible that we enter the sync mode later, so this check should be done within the loop below.

File: src/full_node/full_node.py [hunk 3]
```diff
@@ -860,6 +866,81 @@ async def reject_compact_proof_of_time_request(
         for _ in []:
             yield _
 
+    async def broadcast_uncompact_blocks(
```
Comment (mariano54) [inline src/full_node/full_node.py:869]: Can you add a comment explaining what this method does and the purpose of it?

File: src/full_node/full_node.py [hunk 4]
```diff
@@ -860,6 +866,81 @@ async def reject_compact_proof_of_time_request(
...
+                            block.proof_of_time.number_of_iterations,
+                        )
+                        broadcast_list.append(
+                            (
+                                challenge_msg,
+                                pos_info_msg,
+                            )
+                        )
+                        if (
+                            uncompact_blocks == 0
```
Comment (mariano54) [inline src/full_node/full_node.py:911]: Add comment

File: src/timelord.py [hunk 5]
```diff
@@ -217,6 +221,18 @@ async def _do_process_communication(
         disc: int = create_discriminant(
             challenge_hash, self.constants["DISCRIMINANT_SIZE_BITS"]
         )
+        # Tell the client if we're interested in extending the chain
+        # or to produce compact proofs.
+        if not self.sanitizer_mode:
+            if self.config["fast_algorithm"]:
+                # Run n-wesolowski algorithm.
+                writer.write(b"N")
```
Comment (mariano54) [inline src/timelord.py:229]: This protocol needs to be documented somewhere

Comment (mariano54) [reply src/timelord.py:229]: (unless there is a plan to change it soon)

File: setup.py (hunk 1 above)
Comment (fchirica) [reply setup.py:13]: This is how I used to tag new_code branch (which this code depends on):
https://github.com/Chia-Network/chiavdf/pull/20
We can use normal version numbers once it's merged into master.

**Review by mariano54 — APPROVED** [review]

**Pull request merged**